flink-formats/flink-avro-confluent-registry/src/test/java/org/apache/flink/formats/avro/registry/confluent/RegistryAvroFormatFactoryTest.java

```diff
@@ -229,6 +229,74 @@ public void testSerializationSchemaWithInvalidOptionalSchema() {
                                 null, SCHEMA.toPhysicalRowDataType()));
     }
 
+    @Test
```

Comment:
Nit: Can we turn the three (or at least the first/third) tests into a `parameterTest` to avoid the code duplication?

It's a bit unclear to me why only test two verifies the behavior for sinks and sources while the other tests only look at sinks.